src/event.rs

```diff
@@ -781,6 +781,31 @@ pub struct KeyEvent {
     /// On some systems, holding down a key for some period of time causes that key to be repeated
     /// as though it were being pressed and released repeatedly. This field is `true` if and only if
     /// this event is the result of one of those repeats.
+    ///
+    /// # Example
+    ///
+    /// In games, you often want to ignore repated key events - this can be
+    /// done by ignoring events where this property is set.
+    ///
+    /// ```
+    /// use winit::event::{WindowEvent, KeyEvent, ElementState};
+    /// use winit::keyboard::{KeyCode, PhysicalKey};
+    /// # let window_event = WindowEvent::RedrawRequested; // To make the example compile
+    /// match window_event {
+    ///     WindowEvent::KeyboardInput {
+    ///         event: KeyEvent {
+    ///             physical_key: PhysicalKey::Code(KeyCode::KeyW),
+    ///             state: ElementState::Pressed,
+    ///             repeat: false,
+    ///             ..
```

Comment:
could you include `synthetic: false` here as well?

Reply:
Hmm, do we actually want to recommend that? If I'm writing a 2D game, I'd like to know when the user is pressing `W`, so that I can move the player upwards; in that case, I'd probably _want_ to handle synthetic events too?

Reply:
I mean they are not represent real keyboard input and I'd rather either move them somewhere or remove all together.

Reply:
I've opened https://github.com/rust-windowing/winit/issues/3543 for that.